addons/account_audit_trail/models/account_move.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_account_audit_trail_except_once_post(self):
+        if not self._context.get('force_delete') and any(
+            move.posted_before and move.company_id.check_account_audit_trail
+            for move in self
+        ):
+            raise UserError(_("To keep the audit trail, you can not delete journal entries once they have been posted.\nInstead, you can cancel the journal entry."))
+
+    def unlink(self):
+        # Add logger here because in api ondelete account.move.line is deleted and we can't get total amount
+        logger_msg = False
+        if any(m.posted_before and m.company_id.check_account_audit_trail for m in self):
+            if self._context.get('force_delete'):
+                moves_details = []
+                for move in self:
+                    entry_details = "{entry_number} ({move_id}) amount {amount_total} {currency} and partner {partner_name}".format(
+                        entry_number=move.name,
+                        move_id=move.id,
+                        amount_total=move.amount_total,
+                        currency=move.currency_id.name,
+                        partner_name=move.partner_id.display_name,
+                    )
+                    account_balances_per_account = {}
+                    for line in move.line_ids:
+                        account_id = line.account_id.id
+                        if account_id in account_balances_per_account:
+                            account_balances_per_account[account_id] += line.balance
+                        else:
+                            account_balances_per_account[account_id] = line.balance
+                    for account_id, balance in account_balances_per_account.items():
+                        entry_details += f"\n- {self.env['account.account'].browse(account_id).name} ({account_id}) with balance {balance} {line.currency_id.name}"
```

Comment:
Doesn't that make too many new lines with the `join` below?
```suggestion
                        entry_details += f"- {self.env['account.account'].browse(account_id).name} ({account_id}) with balance {balance} {line.currency_id.name}"
```

Reply:
```\n``` is used to make the lines more readable and to achieve below format

```
Force deleted Journal Entries by Mitchell Admin (2)
Entries
INV/2023/00004 (26) amount 36512.5 USD and partner Deco Addict
- Product Sales (24) with balance -31750.0 USD
- Tax Received (19) with balance -4762.5 USD
- Account Receivable (6) with balance 36512.5 USD
```

Reply:
Sorry, I didn't realize it was appending to a string, too many different ways of programming in 10 lines of code... (`'\n'.join()` vs `+= '\n'`, `"{}".format()` vs `f"{}"`,...)